pkg/proxy/util/utils.go

```diff
@@ -124,6 +125,16 @@ func IsProxyableHostname(ctx context.Context, resolv Resolver, hostname string)
 	return nil
 }
 
+// IsAllowedHost checks if the given IP host address is in a network in the denied list.
+func IsAllowedHost(ctx context.Context, host net.IP, denied []*net.IPNet) error {
```

Comment:
Small quibble but I expect any method with an Is prefix to return a bool

Reply:
I don't disagree, but this is named & used in the same way as IsProxyableHostname which also returns an error rather than a bool. I think it's better to be consistent; the principled alternative would be to change IsProxyableHostname as well which is probably excessive.